internal/pkg/scorecard/scorecard.go

```diff
@@ -273,34 +272,73 @@ func ScorecardTests(cmd *cobra.Command, args []string) error {
 		olmTests.Run(context.TODO())
 		suites = append(suites, olmTests)
 	}
+	return suites, nil
+}
+
+func ScorecardTests(cmd *cobra.Command, args []string) error {
+	if err := initConfig(); err != nil {
+		return err
+	}
+	if err := validateScorecardFlags(); err != nil {
+		return err
+	}
+	cmd.SilenceUsage = true
+	if viper.GetBool(VerboseOpt) {
+		log.SetLevel(logrus.DebugLevel)
+	}
+	suites, err := runTests()
+	if err != nil {
+		return err
+	}
 	totalScore := 0.0
+	// Update the state for the tests
 	for _, suite := range suites {
-		fmt.Printf("%s:\n", suite.GetName())
-		for _, result := range suite.TestResults {
-			fmt.Printf("\t%s: %d/%d\n", result.Test.GetName(), result.EarnedPoints, result.MaximumPoints)
+		for _, res := range suite.TestResults {
+			res.UpdateState()
 		}
-		totalScore += float64(suite.TotalScore())
 	}
-	totalScore = totalScore / float64(len(suites))
-	fmt.Printf("\nTotal Score: %.0f%%\n", totalScore)
-	// Print suggestions
-	for _, suite := range suites {
-		for _, result := range suite.TestResults {
-			for _, suggestion := range result.Suggestions {
-				// 33 is yellow (specifically, the same shade of yellow that logrus uses for warnings)
-				fmt.Printf("\x1b[%dmSUGGESTION:\x1b[0m %s\n", 33, suggestion)
+	if viper.GetString(OutputFormatOpt) == "human-readable" {
+		for _, suite := range suites {
+			fmt.Printf("%s:\n", suite.GetName())
+			for _, result := range suite.TestResults {
+				fmt.Printf("\t%s: %d/%d\n", result.Test.GetName(), result.EarnedPoints, result.MaximumPoints)
 			}
+			totalScore += float64(suite.TotalScore())
 		}
-	}
-	// Print errors
-	for _, suite := range suites {
-		for _, result := range suite.TestResults {
-			for _, err := range result.Errors {
-				// 31 is red (specifically, the same shade of red that logrus uses for errors)
-				fmt.Printf("\x1b[%dmERROR:\x1b[0m %s\n", 31, err)
+		totalScore = totalScore / float64(len(suites))
+		fmt.Printf("\nTotal Score: %.0f%%\n", totalScore)
+		// Print suggestions
+		for _, suite := range suites {
+			for _, result := range suite.TestResults {
```

Comment:
So is the place I am worried about. Technically suite could have a slice in TestResults like:
```
[]{nil, nil, nil}
```

We should really make sure that this is not the case so we don't nil panic here. I would think instead if you don't have the structs of the slice as pointers (which I can't see a reason to have them be a pointer at the moment) then this would not be a concern.